driver/ixsystems/common.py

```diff
@@ -410,45 +410,65 @@ def _create_volume_from_snapshot(self, name, snapshot_name,
-                                                request_urn, None)
-                LOG.debug('_update_volume_stats start /system/version response: %s',versionret)
-                versionresult = json.loads(versionret['response'])
-                LOG.debug('_update_volume_stats /system/version response : %s', versionresult)
-            except Exception as e:
-                raise FreeNASApiError('Unexpected error', e)
-            finally:
-                return str(versionresult)
-            
+        LOG.debug('_update_volume_stats start /system/version request')
+        # Use API v2.0 /system/version to detect nasversion
+        # API v2.0 /system/version available for FreeNAS 11.x TrueNAS 12.x TrueNAS 13.x TrueNAS Scale 22.x
+        request_urn = ("/system/version")
+        self.handle.set_api_version('v2.0')
+        # For legacy verion that does not support API v2.0 /system/version return fallback value "VersionNotFound"
+        versionresult = "VersionNotFound"
+        try:
+            versionret = self.handle.invoke_command(
+                FreeNASServer.SELECT_COMMAND,
+                request_urn, None)
+            LOG.debug('_update_volume_stats start /system/version response: %s', versionret)
+            versionresult = json.loads(versionret['response'])
+            LOG.debug('_update_volume_stats /system/version response : %s', versionresult)
+        except Exception as e:
+            raise FreeNASApiError('Unexpected error', e)
+        finally:
+            return str(versionresult)
+
+    def _tunable(self):
+        LOG.debug('_tunable /tunable request')
+        request_urn = ("/tunable")
+        self.handle.set_api_version('v2.0')
+        tunableresult = []
+        try:
+            tunableret = self.handle.invoke_command(
+                FreeNASServer.SELECT_COMMAND,
+                request_urn, None)
+            tunableresult = json.loads(tunableret['response'])
+            LOG.debug('Tunable response : %s', tunableresult)
+        except Exception as e:
+            raise FreeNASApiError('Unexpected error', e)
+        finally:
+            return tunableresult
+
     def _update_volume_stats(self):
         data = {}
         nasversion = self._system_version()
         # Implementation for TrueNAS 12.0 upwards on API V2.0
         # If user are connecting to FreeNAS report error
-        if nasversion.find("FreeNAS")>=0 or nasversion=="VersionNotFound" :
+        if nasversion.find("FreeNAS") >= 0:
             LOG.error("FreeNAS is no longer support by this version of cinder driver.")
-            raise FreeNASApiError('Version not supported','FreeNAS is no longer support by this version of cinder driver.')
+            raise FreeNASApiError('Version not supported', 'FreeNAS is no longer support by this version of cinder driver.')
+        elif nasversion == "VersionNotFound":
+            LOG.error("TrueNAS not found")
+            raise FreeNASApiError('TrueNAS not found')
```

Comment:
When Truenas server offline, get Truenas api version will return default value, better to handle this case to display proper error message. Old code will display 'Version not supported FreeNAS is no longer support by this version of cinder driver.' which is confusing.